### ISSUE TITLE
Provisional Accounting for Expenses only for Non-Stock Items

### ISSUE THREAD

**Issue by PraveenkumarSubramani** [frappe/erpnext#44831]
### Information about bug

In Purchase, we invoiced a service item, the cost is added under “Cost of Goods Sold”. Here we are not selling that service. How can i nullify that balance under “Cost of Goods Sold”. Provisional Accounting enabled under Company.
As we referred documentation, different accounts used. This was getting confused.

<img width="941" alt="Untitled1" src="https://github.com/user-attachments/assets/c5f92518-f251-4d44-983e-e474e9b8d14f" />
<img width="950" alt="Untitled" src="https://github.com/user-attachments/assets/24845e52-41f6-4497-8341-9d07a1b10be8" />
@ruthra-kumar @rohitwaghchaure 

### Module

accounts, buying

### Version

erpnext 14.70.6
frappe 14.85.1

### Installation method

None

**Comment by prasathchandrasekar** [comment]
bro use Default expense account in item default child table at Item doctypes, which is avoid in the stock-in-hand in sales invoice and cost of goods sold ledger in purchase invoice.
<img width="840" alt="image" src="https://github.com/user-attachments/assets/39280413-9e59-40cb-9a95-ad2ad12164a5" />

**Comment by mihir-kandoi** [comment]
Please provide replication steps.

**Comment by PraveenkumarSubramani** [comment]
1. GL view of Purchase Receipt before Invoice
![image](https://github.com/user-attachments/assets/7ea7fa82-28f5-419e-8fd5-28fdad2ef338)

2. GL view of Purchase Receipt after Invoice
![image](https://github.com/user-attachments/assets/ec86a57d-a60e-4481-9e2e-ee0a0cf19ccd)

3. GL view of Invoice 
![image](https://github.com/user-attachments/assets/1e15d176-a57d-48fa-b37b-cbf059da6b03)

**Comment by PraveenkumarSubramani** [comment]
Once Purchase Invoice Created, reverse entry created for Purchase Receipt and separate entry will getting created for Invoice. In this time Expense Account Credited and Debited in Purchase Receipt. But while invoicing time also it is getting debited. So, balances are added under Expense Account

**Comment by f-9t9it** [comment]
This is still an issue. 

We found it with one of our client. Where COGS is debited at PI as well as PR.

The Wrong entry is 
Purchase Receipt:
COGS Dr. 
Service Received but not billed Cr.

Purchase Invoice:
COGS Dr.  (THIS IS THE WRONG ENTRY)
Creditor Cr

The Correct Entry should be 
Purchase Receipt:
COGS Dr. 
Service Received but not billed Cr.

Purchase Invoice:
Service Received but not billed Dr.
Creditor Cr

The above case is only for Non-Stock Item while Stock Items are following the correct `Stock Received but not billed` Control account in both PR and PI.

I will try to look into the codes and make a pr soon.

**Comment by hans-permana** [comment]
well documented! We are also encountering this issue. Looking forward for the fix.

**Comment by PraveenkumarSubramani** [comment]
@rohitwaghchaure

**Comment by DipenFrappe** [comment]
This is correct behaviour. Actual expenses can only be booked against the purchase invoice, and expenses booked against the purchase receipt are a provision that needs to be reversed.
